src/Playwright/Core/Video.cs

```diff
@@ -35,14 +35,21 @@ internal class Video : IVideo
     public Video(Page page, Connection connection)
     {
         _isRemote = connection.IsRemote;
-        page.Close += (_, _) => _artifactTcs.TrySetCanceled();
-        page.Crash += (_, _) => _artifactTcs.TrySetCanceled();
+        if (page.IsClosed)
+        {
+            _artifactTcs.TrySetResult(null);
+        }
+        page.Close += (_, _) => _artifactTcs.TrySetResult(null);
+        page.Crash += (_, _) => _artifactTcs.TrySetResult(null);
```

Comment:
There is still a potential race condition when the page is closed right after the `page.IsClosed` check and before the event handlers are added, so lets do the check after adding the event handlers.

```suggestion
        page.Close += (_, _) => _artifactTcs.TrySetResult(null);
        page.Crash += (_, _) => _artifactTcs.TrySetResult(null);
        if (page.IsClosed)
        {
            _artifactTcs.TrySetResult(null);
        }
```

Reply:
Note that `IsClosed` is not set when the page crashed, so using `page.ClosedOrCrashedTcs` as in the other PR might be more correct as this would handle closed and crashed correctly.

Reply:
good catch, thanks!